source/fe/mapping_q_eulerian.cc

```diff
@@ -168,34 +168,32 @@ std::vector<Point<spacedim> >
 MappingQEulerian<dim, VectorType, spacedim>::MappingQEulerianGeneric::
 compute_mapping_support_points (const typename Triangulation<dim,spacedim>::cell_iterator &cell) const
 {
-  // first, basic assertion with respect to vector size,
-
   const types::global_dof_index n_dofs  = mapping_q_eulerian.euler_dof_handler->n_dofs();
   const types::global_dof_index vector_size = mapping_q_eulerian.euler_vector->size();
-  (void)n_dofs;
-  (void)vector_size;
 
-  AssertDimension(vector_size,n_dofs);
+  // in case the number of DoFs does not correspond to the size of the
+  // vector, we assume that we are working with a level vector in the
+  // multi-grid context
+  const bool mg_vector = n_dofs != vector_size;
```

Comment:
Is this a safe assumption? I have not really thought about all possibilities, but it seems brittle to me. Isn't there something more generic available at this point?

Reply:
not to my knowledge, but we can extend the interface of `MappingQEulerian` to have additional constructor which accepts `const unsigned int level`, then maybe one can have a level-dofs check here.